[PyTorch] Don't hardcode 4 * Vec::size() in vectorized_reduction

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #138022
* __->__ #138014
* #137722

This will break once we support 128-bit vectors, and there's no reason to do it.

Differential Revision: [D64421982](https://our.internmc.facebook.com/intern/diff/D64421982/)

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

swolchok committed:
[PyTorch] Don't hardcode 4 * Vec::size() in vectorized_reduction

This will break once we support 128-bit vectors, and there's no reason to do it.

Differential Revision: [D64421982](https://our.internmc.facebook.com/intern/diff/D64421982/)

[ghstack-poisoned]

malfet reviewed (approved):
LGTM, just wanted to confirm that `Vec::size()` is constexpr, perhaps adding an explicit constexpr would be a nice touch

swolchok committed:
math is hard on "[PyTorch] Don't hardcode 4 * Vec::size() in vectorized_reduction"


This will break once we support 128-bit vectors, and there's no reason to do it.

Differential Revision: [D64421982](https://our.internmc.facebook.com/intern/diff/D64421982/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
compiling code before you send it out for review is hard on "[PyTorch] Don't hardcode 4 * Vec::size() in vectorized_reduction"


This will break once we support 128-bit vectors, and there's no reason to do it.

Differential Revision: [D64421982](https://our.internmc.facebook.com/intern/diff/D64421982/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge
